File: src/main/java/com/xpdustry/nucleus/metric/MetricSink.java
// SPDX-License-Identifier: GPL-3.0-only
package com.xpdustry.nucleus.metric;

import java.util.Map;

public interface MetricSink {

    default void sample(final String name, final Number value) {
        this.sample(name, value, Map.of());
    }

    void sample(final String name, final Number value, final Map<String, String> labels);
}

File: src/main/kotlin/com/xpdustry/nucleus/metric/MetricSink.kt
// SPDX-License-Identifier: GPL-3.0-only
package com.xpdustry.nucleus.metric

interface MetricSink {
    fun sample(name: String, type: MetricType, value: Number) {
        this.sample(name, type, value, emptyMap())
    }

    fun sample(name: String, type: MetricType, value: Number, labels: Map<String, String>)
}
